File: 2_ArmacodeBeta2/java/MazeyPasscode.java
import java.util.Scanner;

public class PalindromeSubstring {

    public static boolean isPalindrome(String s) {
        for (int i = 0; i < s.length() / 2; i++) {
            if (s.charAt(i) != s.charAt(s.length() - i - 1)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int palindromeLen = scanner.nextInt();
        int len = scanner.nextInt();
        String s = scanner.next();

        for (int i = 0; i <= s.length() - palindromeLen; i++) {
            String substrToCheck = s.substring(i, i + palindromeLen);
            if (isPalindrome(substrToCheck)) {
                System.out.println(substrToCheck);
                break;
            }
        }

        scanner.close();
    }
}

File: 2_ArmacodeBeta2/cpp/MazeyPasscode.cpp
#include <string>
#include <iostream>

using namespace std;

bool isPalindrome(string s) {
    for(unsigned int i = 0; i<s.size()/2; i++) {
        if(s[i] != s[s.size()-i-1]) {
            return false;
        }
    } return true;
}

int main() {
    unsigned int palindromeLen = 0, len = 0;
    cin >> palindromeLen;
    cin >> len;
    string s;
    cin >> s;
    for(unsigned int i = 0; i<s.size()-palindromeLen+1; i++) {
        string substrToCheck = s.substr(i, palindromeLen);
        if(isPalindrome(substrToCheck)) {
            cout << substrToCheck << endl;
            break;
        }
    }
    return 0;
}